internal/watcher/instance/instance_watcher_service.go

```diff
@@ -263,7 +261,27 @@ func areInstancesEqual(oldRuntime, currentRuntime *mpi.InstanceRuntime) (equal b
 		return false
 	}
 
-	return reflect.DeepEqual(oldRuntime.GetInstanceChildren(), currentRuntime.GetInstanceChildren())
+	oldRuntimeChildren := oldRuntime.GetInstanceChildren()
+	currentRuntimeChildren := currentRuntime.GetInstanceChildren()
+
+	if len(oldRuntimeChildren) != len(currentRuntimeChildren) {
+		return false
+	}
+
+	for _, oldChild := range oldRuntimeChildren {
+		childFound := false
+		for _, currentChild := range currentRuntimeChildren {
+			if oldChild.GetProcessId() == currentChild.GetProcessId() {
+				childFound = true
+			}
+		}
+
+		if !childFound {
```

Comment:
return childFound is ok on it's own?

Reply:
no if its true I dont want to return. I want to check if the next child is found